extensions/cornerstone-dicom-sr/src/getSopClassHandlerModule.js

```diff
@@ -221,11 +221,23 @@ function _checkIfCanAddMeasurementsToDisplaySet(
     const image = images[i];
     const { SOPInstanceUID } = image;
     if (SOPInstanceUIDs.includes(SOPInstanceUID)) {
-      const imageId = imageIdsForDisplaySet[i];
-
       for (let j = unloadedMeasurements.length - 1; j >= 0; j--) {
         const measurement = unloadedMeasurements[j];
         if (_measurementReferencesSOPInstanceUID(measurement, SOPInstanceUID)) {
```

Comment:
I think I would include the frame number extracted from the imageId here, as _measurementReferencesSOPInstanceUID(measurement, SOPInstanceUID, frame#), as the code below ends up checking for some things that may not be true.  Specifically:
1. ReferencedFrameNumber may be an array of values
2. Frames do not need to be in order, nor all in the same display set.

The original logic, just say does a measurement reference this image id should work for all of them, as long as you extract both the sop instance uid and the frame#

Reply:
This needs to be done by using the MetadataProvider to call the _getUIDsFromImageId (you will have to make it public), and then do the comparison on that returned data, as there is no guarantee that a single frame series doesn't have multiframe data in it, or that a multiframe series doesn't have two or more sop instances in it (they can get broken up for very large data sets).

Reply:
Fixed as discussed in https://github.com/OHIF/Viewers/pull/2973/commits/3ee38072d4309c1524a22a4da94bc41ee7bf4bd8 and https://github.com/OHIF/Viewers/pull/2973/commits/75b4cfcef27b972745ba2e3ea3f9546d5f8cf27d